etl/stores/arkg_store.py

```diff
@@ -0,0 +1,88 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Self
+
+import pyoxigraph as ox
+
+from etl.models import AntiRecommendationGraphTuple
+from etl.models.types import RdfMimeType
+from etl.pipelines.arkg_builder_pipeline import ArkgBuilderPipeline
+
+
+class ArkgStore:
+    """A database to store an Anti-Recommendation Knowledge Graph."""
+
+    @dataclass(frozen=True)
+    class Descriptor:
+        """A dataclass that holds the Path of the resource stored in database."""
+
+        arkg_store_path: Path
+
+    def __init__(self, *, arkg_store: ox.Store, arkg_store_path: Path) -> None:
+        self.__arkg_store = arkg_store
+        self.__arkg_store_path = arkg_store_path
+
+    def __enter__(self) -> Self:
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:  # noqa: ANN001
+        del self.__arkg_store
+
+    @classmethod
+    def create(
+        cls,
+        *,
+        arkg_store_path: Path,
+        requests_cache_directory: Path,
+        anti_recommendation_graphs: AntiRecommendationGraphTuple,
+    ) -> Self:
+        """
+        Return an ArkgDatabase that contains an ARKG Store constructed with an ArkgBuilderPipeline.
+        """
+
+        return cls(
+            arkg_store=ArkgBuilderPipeline(
+                arkg_store_directory_path=arkg_store_path,
+                requests_cache_directory=requests_cache_directory,
+            ).construct_graph(anti_recommendation_graphs.anti_recommendation_graphs),
+            arkg_store_path=arkg_store_path,
+        )
+
+    @classmethod
+    def open(cls, descriptor: Descriptor) -> Self:
+        """Return an ArkgDatabase that has an empty RDF Store, and an arkg_file_path that is set to descriptor."""
+
+        return cls(
+            arkg_store=ox.Store(path=descriptor.arkg_store_path),
+            arkg_store_path=descriptor.arkg_store_path,
+        )
+
+    @property
+    def arkg_store(self) -> ox.Store:
```

Comment:
Why are you exposing this store? Since you encapsulate it in `dump` and `load`.

Reply:
I have encapsulated the store in ArkgStore.